changes/2020-07-15_specify-behavior-for-raw-aes-keyring-for-invalid-EC/change.md

```diff
@@ -0,0 +1,99 @@
+
+## Affected Features
+
+This serves as a reference of all features that this change affects.
+
+| Feature                                               |
+| ----------------------------------------------------- |
+| [raw-aes-keyring](../../framework/raw-aes-keyring.md) |
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                                         |
+| ----------------------------------------------------- |
+| [raw-aes-keyring](../../framework/raw-aes-keyring.md) |
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+## Definitions
+
+- "serialize", "serialization", and "serializability" all refer to
+  [message header serialization of AAD key value pairs](../../data-format/message-header.md#key-value-pairs)
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The raw-aes-keyring must serialize the encryption context for use as AAD.
+The specification did not describe how the keyring should behave when given an unserializable encryption context.
+
+With this change, the raw-aes-keyring MUST attempt to serialize the encryption context,
+prior to attempting encryption/decryption.
+If the raw-aes-keyring cannot serialize the encryption context, it MUST fail.
+
+## Out of Scope
+
+- All keyrings aside from the raw-aes-keyring
+- The encryption context serialization format
+
+## Motivation
+
+The C and Javascript implementations of the raw-aes-keyring check for serializability at different points in OnDecrypt.
+This has potential to lead to different behavior under specific circumstances.
+
+## Drawbacks
+
+Since we attempt to serialize the encryption context before attempting any decryption,
+the raw-aes-keyring may attempt to serialize an encryption context in cases where no EDK matches the keyring.
+This scenario results in slower operation when compared to the current Javascript implementation,
+which only checks serializability when attempting to decrypt an EDK.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will improve behavioral consistency between raw-aes-keyring implications.
+This should improve customer experience, and reduce support engagements.
```

Comment:
np: `This should improve customer experience, and reduce support engagements.` -> `This should improve the customer experience and reduce support engagements.`